### PR TITLE
fix: golint, gosec

### PR DESCRIPTION
**Pull request by psturc**
[HACBS-1055](https://issues.redhat.com//browse/HACBS-1055)

a follow-up PR on https://github.com/redhat-appstudio/jvm-build-service/pull/200

### CODE REVIEW

File: openshift-with-appstudio-test/e2e/jvm_build_service_test.go [hunk 1]
```diff
@@ -341,7 +341,7 @@ func TestExampleRun(t *testing.T) {
 							ta.Logf(fmt.Sprintf("error getting pod logs for container %s: %s", container.Name, err.Error()))
 							continue
 						}
-						b, err := ioutil.ReadAll(readCloser)
+						b, err := io.ReadAll(readCloser)
```
Comment (psturc) [inline openshift-with-appstudio-test/e2e/jvm_build_service_test.go:344]: [ioutil deprecation](https://pkg.go.dev/io/ioutil#pkg-overview)

File: openshift-with-appstudio-test/e2e/util.go [hunk 2]
```diff
@@ -264,7 +262,7 @@ func decodeBytesToTektonObjbytes(bytes []byte, obj runtime.Object, ta *testArgs)
 }
 
 func streamRemoteYamlToTektonObj(url string, obj runtime.Object, ta *testArgs) runtime.Object {
-	resp, err := http.Get(url)
+	resp, err := http.Get(url) //#nosec G107
```
Comment (psturc) [inline openshift-with-appstudio-test/e2e/util.go:265]: here the gosec warned about "Potential HTTP request made with variable url"

File: openshift-with-appstudio-test/e2e/util.go [hunk 3]
```diff
@@ -277,7 +275,7 @@ func streamRemoteYamlToTektonObj(url string, obj runtime.Object, ta *testArgs) r
 }
 
 func streamFileYamlToTektonObj(path string, obj runtime.Object, ta *testArgs) runtime.Object {
-	bytes, err := os.ReadFile(path)
+	bytes, err := os.ReadFile(filepath.Clean(path))
```
Comment (psturc) [inline openshift-with-appstudio-test/e2e/util.go:278]: gosec warning: ["Potential file inclusion via variable".](https://securego.io/docs/rules/g304.html)

Fixed

File: openshift-with-appstudio-test/e2e/util.go [hunk 4]
```diff
@@ -321,6 +319,7 @@ func dumpPodsGlob(ta *testArgs, namespace, glob string) {
 	}
 }
 
+/*
```
Comment (psturc) [inline openshift-with-appstudio-test/e2e/util.go:322]: unused function

File: openshift-with-appstudio-test/e2e/client.go [hunk 5]
```diff
@@ -1,6 +1,3 @@
-//go:build normal || periodic
```
Comment (psturc) [inline openshift-with-appstudio-test/e2e/client.go:1]: @gabemontero I messed up in the previous PR. These tags were not needed. What needed an update was the [gosec and golangci-lint command args](https://github.com/redhat-appstudio/jvm-build-service/pull/207/files#diff-b3ff1ee6de56e14f44fca2bf15023e30ac069e3daaf0cf204d9c4f94712974f8)

**Comment by psturc** [comment]
[ginkgo bootstrap failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/207/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1564909528519020544) 

/test jvm-build-service-in-repo-e2e

**Comment by psturc** [comment]
* jvm-in-repo-test: [must gather step has failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/207/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1564922488037576704) 😑 
* "e2e-tests repo e2e" failed on
  ```
  [ERROR] Plugin org.apache.maven.plugins:maven-shade-plugin:3.2.4 or one of its dependencies could not be resolved: 
  Failed to read artifact descriptor for org.apache.maven.plugins:maven-shade-plugin:jar:3.2.4: Could not transfer artifact 
  org.apache.maven.plugins:maven-shade-plugin:pom:3.2.4 from/to mirror.default (http://localhost:2000/maven2): Transfer 
  failed for http://localhost:2000/maven2/org/apache/maven/plugins/maven-shade-plugin/3.2.4/maven-shade-plugin- 
  3.2.4.pom: Connect to localhost:2000 [localhost/127.0.0.1] failed: Connection refused -> [Help 1]
  ```
  [link to full log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/207/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-e2e/1564909528485466112/artifacts/jvm-build-service-e2e/jvm-build-service-e2e/artifacts/jvm-build-service-test/jvm-build-uwpm-pr-jvm-build-suite-component-klnm-rbknl.log)
  - @gabemontero do you think we can do something about such failure?

/test jvm-build-service-e2e
/test jvm-build-service-in-repo-e2e

**Comment by psturc** [comment]
CI issue
```
Unable to connect to the server: dial tcp: lookup api.ci-ocp-4-10-amd64-aws-us-east-1-wcjr9.hive.aws.ci.openshift.org on 172.30.0.10:53: no such host
```
/test jvm-build-service-in-repo-e2e

**Comment by psturc** [comment]
openshift-ci connection issues again

/test jvm-build-service-in-repo-e2e

**Comment by gabemontero** [comment]
> * jvm-in-repo-test: [must gather step has failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/207/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e/1564922488037576704) expressionless
> 
>     * "e2e-tests repo e2e" failed on
>       ```
>       [ERROR] Plugin org.apache.maven.plugins:maven-shade-plugin:3.2.4 or one of its dependencies could not be resolved: 
>       Failed to read artifact descriptor for org.apache.maven.plugins:maven-shade-plugin:jar:3.2.4: Could not transfer artifact 
>       org.apache.maven.plugins:maven-shade-plugin:pom:3.2.4 from/to mirror.default (http://localhost:2000/maven2): Transfer 
>       failed for http://localhost:2000/maven2/org/apache/maven/plugins/maven-shade-plugin/3.2.4/maven-shade-plugin- 
>       3.2.4.pom: Connect to localhost:2000 [localhost/127.0.0.1] failed: Connection refused -> [Help 1]
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       [link to full log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/207/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-e2e/1564909528485466112/artifacts/jvm-build-service-e2e/jvm-build-service-e2e/artifacts/jvm-build-service-test/jvm-build-uwpm-pr-jvm-build-suite-component-klnm-rbknl.log)
>       
>       * @gabemontero do you think we can do something about such failure?

more @stuartwdouglas 's domain @psturc .... this has been popping up intermittently in various PR e2e's

he mentioned in slack he suspects it is a new set of memory - see https://coreos.slack.com/archives/C03KN2B47GU/p1661893679463399

and while I've seen it in any of the e2e's, I suspect we see it more often in the component based test because it pulls in the hacbs bundle version of things, which I think adds more tasks onto the java build (hence increasing the memory demand)
> 
> 
> /test jvm-build-service-e2e /test jvm-build-service-in-repo-e2e

**Comment by gabemontero** [comment]
@psturc @stuartwdouglas this sidecar container log from the failed in repo e2e was interesting / one I had not seen before:

```
    types.go:25: time: 2022-08-31 13:25:21.394598725 +0000 UTC m=+2229.583068374: pod logs for container sidecar-proxy in pod sample-component-build-dtqvr-maven-run-pod:  INFO exec  java -Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager -Xms64m -Xmx256m -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:+ExitOnOutOfMemoryError -cp "." -jar /deployments/quarkus-run.jar 
        __  ____  __  _____   ___  __ ____  ______ 
         --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
         -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
        --\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
        2022-08-31 13:02:53,235 INFO  [io.quarkus] (main) hacbs-sidecar 1.0-SNAPSHOT on JVM (powered by Quarkus 2.7.5.Final) started in 3.207s. Listening on: http://0.0.0.0:2000
        2022-08-31 13:02:53,424 INFO  [io.quarkus] (main) Profile prod activated. 
        2022-08-31 13:02:53,425 INFO  [io.quarkus] (main) Installed features: [amazon-s3, cdi, rest-client-reactive, resteasy-reactive, smallrye-context-propagation, smallrye-health, vertx]
        2022-08-31 13:02:57,216 INFO  [com.red.hac.sid.res.MavenProxyResource] (executor-thread-1) Constructing resource manager with build policy default
        2022-08-31 13:03:38,827 ERROR [io.qua.ver.cor.run.VertxCoreRecorder] (vert.x-eventloop-thread-1) Uncaught exception received by Vert.x: java.lang.IllegalStateException: Can't set the context safety flag: the current context is not a duplicated context
        	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.setContextSafe(VertxContextSafetyToggle.java:107)
        	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.setCurrentContextSafe(VertxContextSafetyToggle.java:94)
        	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$12.handle(VertxHttpRecorder.java:504)
        	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$12.handle(VertxHttpRecorder.java:501)
        	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:161)
        	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:143)
        	at io.vertx.core.http.impl.Http1xServerRequestHandler.handle(Http1xServerRequestHandler.java:67)
        	at io.vertx.core.http.impl.Http1xServerRequestHandler.handle(Http1xServerRequestHandler.java:30)
        	at io.vertx.core.http.impl.Http1xServerConnection.lambda$handleNext$1(Http1xServerConnection.java:225)
        	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:50)
        	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:274)
        	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:22)
        	at io.vertx.core.http.impl.Http1xServerConnection.handleNext(Http1xServerConnection.java:222)
        	at io.vertx.core.http.impl.Http1xServerConnection.responseComplete(Http1xServerConnection.java:207)
        	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
        	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
        	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
        	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
        	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by gabemontero** [comment]
> 

talked to @stuartwdouglas about ^^ last night in slack @psturc and he said this was a quarkus bug; that said, given his history with quarkus, not surprisingly :-), he thought he could fix it upstream.  Definitely feels intermittent.  

All that said, the hoopla that arose yesterday / last night with the gradle build kept him and @dwalluck pretty busy, so I suspect fixing quarkus got put on the back burner, at least for the short term.

**Pull request merged**